File: app/src/main/java/com/saee/combattimer/ui/theme/CombatColors.kt
package com.saee.combattimer.ui.theme

import androidx.compose.ui.graphics.Color

/**
 * Shonen-manga fight-poster palette: deep electric blue, punchy glove red,
 * bright impact yellow, and thick black ink outlines on everything — the
 * Hajime no Ippo cover-art look, not a muted/vintage one.
 */
object CombatColors {
    val Ink = Color(0xFF0A0A0A)
    val Night = Color(0xFF0B1020)
    val PanelNavy = Color(0xFF121A33)
    val PanelNavyAlt = Color(0xFF1A2547)

    val BurstBlue = Color(0xFF1746C4)
    val BurstBlueLight = Color(0xFF4B7BFF)
    val BurstBlueDark = Color(0xFF0C2A80)

    val GloveRed = Color(0xFFE31E24)
    val GloveRedDark = Color(0xFF8E1216)

    val ImpactYellow = Color(0xFFFFD400)
    val ImpactYellowDark = Color(0xFFB88A00)

    val Paper = Color(0xFFF5F1E6)

    // Backwards-compatible aliases used by phase logic elsewhere in the app.
    val InkBlack = Night
    val PanelBlack = PanelNavy
    val PanelBlackAlt = PanelNavyAlt
    val RopeGrey = Ink
    val CornerRed = GloveRed
    val CornerRedDim = GloveRedDark
    val CornerAmber = ImpactYellow
    val CornerAmberDim = ImpactYellowDark
    val CornerGreen = BurstBlue
    val CornerGreenDim = BurstBlueDark
    val Bone = Paper
}

File: ios/CombatTimer/Theme/CombatColors.swift
import SwiftUI

/// Shonen-manga fight-poster palette: deep electric blue, punchy glove red,
/// bright impact yellow, and thick black ink outlines — matches the
/// Android app's theme exactly.
enum CombatColors {
    static let ink = Color(hex: 0x0A0A0A)
    static let night = Color(hex: 0x0B1020)
    static let panelNavy = Color(hex: 0x121A33)
    static let panelNavyAlt = Color(hex: 0x1A2547)

    static let burstBlue = Color(hex: 0x1746C4)
    static let burstBlueLight = Color(hex: 0x4B7BFF)
    static let burstBlueDark = Color(hex: 0x0C2A80)

    static let gloveRed = Color(hex: 0xE31E24)
    static let gloveRedDark = Color(hex: 0x8E1216)

    static let impactYellow = Color(hex: 0xFFD400)
    static let impactYellowDark = Color(hex: 0xB88A00)

    static let paper = Color(hex: 0xF5F1E6)
}

extension Color {
    init(hex: UInt32) {
        let red = Double((hex >> 16) & 0xFF) / 255
        let green = Double((hex >> 8) & 0xFF) / 255
        let blue = Double(hex & 0xFF) / 255
        self.init(red: red, green: green, blue: blue)
    }
}
